### PR TITLE
Fix Validate JSON Schema Test case

### PR DESCRIPTION
**Pull request by Nirothipan**
## Purpose

Include fixes from https://github.com/wso2/carbon-mediation/pull/1270 which fixes 
 wso2/micro-integrator#180

The metadata related to registry resource is stored in a separate file named <resource_file_name>.meta inside the ".metadata" dir which is located where the registry resource is.
At the moment media type is the only metadata stored.

Example:
```
|registry/
| ---- | governance/
| -------- | custom1/
| ------------ | checkJsScript.js
| ------------ | .metadata/
| ---------------- | checkJsScript.js.meta
```

Fixes ValidateJSONSchemaTestCase .

### CODE REVIEW

**Comment by milindaperera** [comment]
This is improvement to file based registry, not fixing the test case, hence shall we update description with the improvement made

**Comment by Nirothipan** [comment]
> This is improvement to file based registry, not only fixing the test case, hence shall we update description with the improvement made

done .

File: components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java [hunk 1]
```diff
@@ -220,6 +225,24 @@ public OMNode lookup(String key) {
...
+    private boolean lookupUtil(String key, URL url) {
+        if (url == null) {
+            handleException("Unable to create URL for target resource : " + key);
+        }
+
+        if ("file".equals(url.getProtocol())) {
+            try {
+                url.openStream();
+            } catch (IOException e) {
+                if (log.isDebugEnabled()) {
```
Comment (sdkottegoda) [inline components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java]: Should this be an error log?

Comment (Nirothipan) [reply components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java]: This is just registry lookup and if the resource is not there we are returning null , hence I don't think this need to be an error log.

Comment (sdkottegoda) [reply components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java]: but is it correct to say that an IOException would be thrown only for a missing file?

Comment (Nirothipan) [reply components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java]: added in 0ee8df54ff7fa587de65c6d09435573d5c8f59f2

**Pull request merged**